### PR TITLE
Allow to await the sidebar

### PR DESCRIPTION
**Pull request by skjnldsv**
Allow us to await or use then/catch when requesting a file to the sidebar

### CODE REVIEW

File: apps/files/src/sidebar.js [hunk 1]
```diff
@@ -51,10 +51,10 @@ window.addEventListener('DOMContentLoaded', () => {
...
 	// Init vue app
-	const AppSidebar = new Vue({
-		// eslint-disable-next-line vue/match-component-file-name
+	const View = Vue.extend(SidebarView)
+	const AppSidebar = new View({
 		name: 'SidebarRoot',
-		render: h => h(SidebarView),
 	})
 	AppSidebar.$mount('#app-sidebar')
+	window.OCA.Files.Sidebar.open = AppSidebar.open
```
Comment (juliushaertl) [inline apps/files/src/sidebar.js]: close is missing, also we currently use the following in the file actions to make sure that the sidebar is properly updated if opened already:

https://github.com/nextcloud/server/blob/16e0887ec63591113ee3f476e0c5129e20180cde/apps/files/js/fileactions.js#L710

Comment (skjnldsv) [reply apps/files/src/sidebar.js]: Fixed

**Review by juliusknorr — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by rullzer** [comment]
Needs a rebase!

**Comment by skjnldsv** [comment]
@juliushaertl & I approved the backport as this is mostly some code moving around.
@ChristophWurst is against it (that would greatly improve the viewer handling of the sidebar :wink: )

**Comment by ChristophWurst** [comment]
> @ChristophWurst is against it (that would greatly improve the viewer handling of the sidebar wink )

@nickvergessen @rullzer your call. We usually don't backport feature or API changes.

**Pull request merged**